src/main/java/core/basesyntax/impl/AmountFromFileImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.db.FruitShopDao;
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.service.AmountFromFile;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AmountFromFileImpl implements AmountFromFile {
+    private FruitShopDao fruitShopDao;
+
+    public AmountFromFileImpl(FruitShopDao fruitShopDao) {
+        this.fruitShopDao = fruitShopDao;
+    }
+
+    @Override
+    public List<Integer> getAmountInShop(List<String> fruits,
+                                         List<FruitTransaction> fruitTransactions) {
+        List<Integer> amounts = new ArrayList<>();
+
+        for (int i = 0; i < fruits.size(); i++) {
+            String fruit = fruits.get(i);
+            int sum = fruitTransactions.stream()
+                    .filter(ft -> ft.getFruit().equals(fruit))
+                    .mapToInt(n -> calculateQantityFromOperation(n))
+                    .sum();
+            amounts.add(sum);
+            fruitShopDao.put(fruit,sum);
+        }
+        return amounts;
+    }
+
+    private static int calculateQantityFromOperation(FruitTransaction transaction) {
+        return transaction.getOperation() == FruitTransaction.Operation.BALANCE
```

Comment:
The use of multiple ternary operators makes this code hard to read and maintain. Consider refactoring this to use a strategy pattern for handling different operations, which will improve readability and extensibility.